src/backend/api/meals.js

```diff
@@ -2,13 +2,119 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
-router.get("/", async (request, response) => {
+  router.get("/", async (req, res) => {
+    //week3
+    const { maxPrice, availableReservations, title, dateAfter, dateBefore, limit, sortKey,} =
+    req.query;
+  
+    let sortDir = req.query.sortDir;
+
+    const response = {
+      data: [],
+      status: 200,
+      message: "the message",
+    };
+
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+
+    // api/meals?maxPrice=90
+    if (maxPrice) {
+      const result = await knex
+        .select("*")
+        .from("meal")
+        .where("price", "<", +maxPrice);
+
+      response.data = result;
+    }
+
+    //api/meals?availableReservations=true
+    if (availableReservations===true) {
+      const result = await knex
+      .select('meal_id', 'meal_title', 'max_reservations', knex
+      .raw('-SUM(number_of_guests)'), knex
+      .raw('max_reservations - SUM(number_of_guests)'))
+      .from('meal')
+      .leftJoin('reservation', 'meal_id', 'reservation.meal_id')
+      .groupBy('meal_id')
+      .having(knex.raw('SUM(number_of_guests) < max_reservations'));
+
+      response.data = result;
+    } else {
+      result = await knex
+      .select('meal_id', 'meal_title', 'max_reservations', knex
+      .raw('-SUM(number_of_guests)'), knex
+      .raw('max_reservations - SUM(number_of_guests)'))
+      .from('meal')
+      .leftJoin('reservation', 'meal_id', 'reservation.meal_id')
+      .groupBy('meal_id')
+      .having(knex.raw('SUM(number_of_guests) > max_reservations'));
+
+      response.data = result;
+    }
+
+    //api/meals?title=Rød%20grød
+    if (title) {
+      const result = await knex("meals")
+      .select("*")
+      .where("title", "like", `%${title}%`);
+
+      response.data = result;
+    }
+
+    //api/meals?dateAfter=2022-10-01
+    if (dateAfter) {
+      const result = await knex
+        .select("*")
+        .from("meals")
+        .where("meal_time", ">", new Date(dateAfter));
+
+      response.data = result;
+    }
+
+    //api/meals?dateBefore=2022-08-08
+    if (dateBefore) {
+      const result = await knex("meals")
+      .select("*")
+      .where("meal_time", "<", dateBefore);
+
+      response.data = result;
+    }
+
+    //api/meals?limit=7
+    if (limit) {
+      const result = await knex("meal")
+        .select("*")
+        .limit(+limit);
+
+      response.data = result;
+    }
+
+    //api/meals?sortKey=price
+    if (["meal_time", "max_reservations", "price"].includes(sortKey)) {
+      const result = await knex("meal")
+        .select("*")
+        .orderBy(sortKey, sortDir === "desc" ? sortDir : "asc");
```

Comment:
I see that here you use not only the `sortKey` parameter, but the `sortDir` too.